src/sys/unix.rs

```diff
@@ -1,52 +1,55 @@
 use libc;
-use std::{mem, slice, usize};
+use std::{slice, usize};
 
+#[derive(Clone)]
 pub struct IoVec {
-    inner: [u8],
+    inner: libc::iovec,
 }
 
 pub const MAX_LENGTH: usize = usize::MAX;
 
 impl IoVec {
     pub fn as_ref(&self) -> &[u8] {
         unsafe {
-            let vec = self.iovec();
-            slice::from_raw_parts(vec.iov_base as *const u8, vec.iov_len)
+            slice::from_raw_parts(
+                self.inner.iov_base as *const u8,
+                self.inner.iov_len)
         }
     }
 
     pub fn as_mut(&mut self) -> &mut [u8] {
         unsafe {
-            let vec = self.iovec();
-            slice::from_raw_parts_mut(vec.iov_base as *mut u8, vec.iov_len)
+            slice::from_raw_parts_mut(
+                self.inner.iov_base as *mut u8,
+                self.inner.iov_len)
         }
     }
-
-    unsafe fn iovec(&self) -> libc::iovec {
-        mem::transmute(&self.inner)
-    }
 }
 
-impl<'a> From<&'a [u8]> for &'a IoVec {
+impl<'a> From<&'a [u8]> for IoVec {
```

Comment:
This should probably not go through `From` and instead be marked as `unsafe` as it's actually unsafe